src/Razor/Razor/src/IRazorRazorPage.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Html;
+
+namespace Microsoft.AspNetCore.Razor
+{
+    /// <summary>
+    /// Represents properties and methods that are used by a Razor View for execution.
+    /// </summary>
+    public interface IRazorRazorPage<TOutputContext> where TOutputContext : IOutputContext
```

Comment:
I'm not sure that there's much value in putting this stuff in an interface

Reply:
It would be nice to have at least some common interface for the code that the code generated by razor calls, instead of relying on convention, and keeping two separate classes in-sync. Especially if customers are allowed to implement and specify their own base classes.